### ISSUE TITLE
6.x.x - Foods And Drinks

### ISSUE THREAD

**Issue by silvermcd123** [TrinityCore/TrinityCore#14112]
REV: https://github.com/TrinityCore/TrinityCore/commit/c4d81aa42d4918fa6a9b86bfea017f36192155ed

The issue: cannot drink or eat to recover hp/mana

**Comment by Epicurus4** [comment]
```CMSG_USE_ITEM``` needs to be scripted, and even then the item will only recover 19 or so health. There is a broader issue with the base spell system of TrinityCore which will prevent the hp/mana recovery amount from triggering the correct value. Fixing the struct for ```SMSG_SPELL_GO``` and ```SMSG_SPELL_START``` does not correct the recovered amount either - so there is an underlying issue in the spell system itself which causes this.

For the devs reading this, I did not make a PR for my fixed ```SMSG_SPELL_GO``` and ```SMSG_SPELL_START``` structs because I wrote them using the old packet system.
